System/Directory/Internal/Posix.hsc

```diff
@@ -146,7 +146,13 @@ linkToDirectoryIsDirectory :: Bool
 linkToDirectoryIsDirectory = False
 
 createHardLink :: OsPath -> OsPath -> IO ()
+#ifdef haiku_HOST_OS
+-- Haiku's default file system, BFS, does not support hard links
+createHardLink (OsString p1) (OsString p2) =
+  Posix.createSymbolicLink p1 p2
```

Comment:
This may be risky. Replacing hard link with symbolic links can alter the semantics of programs that were expecting hard links.